lib/FrontendTool/FrontendTool.cpp

```diff
@@ -219,38 +239,79 @@ static bool emitLoadedModuleTraceIfNeeded(ASTContext &ctxt,
     return true;
   }
 
-  llvm::SmallVector<std::string, 16> swiftModules;
+  ModuleDecl* currentModule = ctxt.getModuleByName(moduleName);
```

Comment:
The calling context has this (possibly several levels up). We should just pass it in if it's important. (Maybe instead of the ASTContext, since every ModuleDecl contains a reference to the ASTContext for convenience.)

Also, `*` attached to the variable name, please, per C parsing rules.

Reply:
> The calling context has this (possibly several levels up). We should just pass it in if it's important.

Got it.

> Also, * attached to the variable name, please, per C parsing rules.

I slipped up (old habits, ugh). I recall you'd mentioned this earlier as well. Will fix.

Reply:
Looking back at this, the callers don't have a hold on the ModuleDecl, and passing the Context is important (I think), because we are using it to emit errors in some cases. If we instead pass the ModuleDecl and then potentially start emitting errors using the context we obtain through it (instead of explicitly passing it in), that seems a bit odd to me. It's somewhat like passing in the root of a tree (ASTContext) vs a leaf with a back pointer (ModuleDecl).

Reply:
The callers have the ModuleDecl; it's in the CompilerInstance (as the "main module"). And I see your point, but ASTContext isn't the root of a tree (since you can't walk it…or at least aren't expected to); it's the substrate the tree is made up of. Or something like that. Passing many parameters is also a readability tradeoff.